model/multi_doc/base_model.py

```diff
@@ -0,0 +1,29 @@
+from model.base_model import SummModel
+from typing import List, Union
+
+
+class MultiDocSummModel(SummModel):
+    def __init__(self,
+                 trained_domain: str = None,
+                 max_input_length: int = None,
+                 max_output_length: int = None,
+                 ):
+        super(MultiDocSummModel, self).__init__(trained_domain = trained_domain, max_input_length = max_input_length, max_output_length = max_output_length)
+
+    @classmethod
+    def assert_summ_input_type(cls, corpus, query):
+        for instance in corpus:
```

Comment:
you don't have to use `for` loop here, you can do `if not all([isinstance(ins, list) for ins in corpus]): ...`

similarly, you can drop the nested loops and do `if not all([isinstance(doc, str) for doc in ins for ins in corpus])